checker/jdk/signedness/src/java/lang/Integer.java

```diff
@@ -0,0 +1,1598 @@
+     * no-op, so all but the last five bits of the rotation distance can be
+     * ignored, even if the distance is negative: {@code rotateRight(val,
+     * distance) == rotateRight(val, distance & 0x1F)}.
+     *
+     * @param i the value whose bits are to be rotated right
+     * @param distance the number of bit positions to rotate right
+     * @return the value obtained by rotating the two's complement binary
+     *     representation of the specified {@code int} value right by the
+     *     specified number of bits.
+     * @since 1.5
+     */
+    public static @PolySigned int rotateRight(@PolySigned int i, int distance) {
+        return (i >>> distance) | (i << -distance);
+    }
+
+    /**
+     * Returns the value obtained by reversing the order of the bits in the
+     * two's complement binary representation of the specified {@code int}
+     * value.
+     *
+     * @param i the value to be reversed
+     * @return the value obtained by reversing order of the bits in the
+     *     specified {@code int} value.
+     * @since 1.5
+     */
+    public static @SignednessEither int reverse(@UnknownSignedness int i) {
+        // HD, Figure 7-1
+        i = (i & 0x55555555) << 1 | (i >>> 1) & 0x55555555;
+        i = (i & 0x33333333) << 2 | (i >>> 2) & 0x33333333;
+        i = (i & 0x0f0f0f0f) << 4 | (i >>> 4) & 0x0f0f0f0f;
+        i = (i << 24) | ((i & 0xff00) << 8) |
+            ((i >>> 8) & 0xff00) | (i >>> 24);
+        return i;
+    }
+
+    /**
+     * Returns the signum function of the specified {@code int} value.  (The
+     * return value is -1 if the specified value is negative; 0 if the
+     * specified value is zero; and 1 if the specified value is positive.)
+     *
+     * @param i the value whose signum is to be computed
+     * @return the signum function of the specified {@code int} value.
+     * @since 1.5
+     */
+    public static int signum(int i) {
+        // HD, Section 2-7
+        return (i >> 31) | (-i >>> 31);
+    }
+
+    /**
+     * Returns the value obtained by reversing the order of the bytes in the
+     * two's complement representation of the specified {@code int} value.
+     *
+     * @param i the value whose bytes are to be reversed
+     * @return the value obtained by reversing the bytes in the specified
+     *     {@code int} value.
+     * @since 1.5
+     */
+    public static @SignednessEither int reverseBytes(@UnknownSignedness int i) {
+        return ((i >>> 24)           ) |
+               ((i >>   8) &   0xFF00) |
+               ((i <<   8) & 0xFF0000) |
+               ((i << 24));
+    }
+
+    /**
+     * Adds two integers together as per the + operator.
+     *
+     * @param a the first operand
+     * @param b the second operand
+     * @return the sum of {@code a} and {@code b}
+     * @see java.util.function.BinaryOperator
+     * @since 1.8
+     */
+    public static @SignednessEither int sum(@PolySigned int a, @PolySigned int b) {
```

Comment:
This annotation is incorrect.  The sum of two arbitrary values is not guaranteed to be in the signed positive range.

You can verify this by type-checking the following program:

```
import org.checkerframework.checker.signedness.qual.SignednessEither;
import org.checkerframework.checker.signedness.qual.PolySigned;

public class Sum {

  public static @SignednessEither int sum(@PolySigned int a, @PolySigned int b) {
         return a + b;
    }
}
```

which yields the following output:

```
javac -processor signedness Sum.java
Sum.java:7: error: [return.type.incompatible] incompatible types in return.
         return a + b;
                  ^
  type of expression: @PolySigned int
  method return type: @SignednessEither int
1 error
```

Could you please:

1. Determine why the Signedness Checker issued no warning when compiling the annotated JDK, and ensure that it does so.
2. Only after doing that, fix this error (and any others that the type-checker issues while compiling the Signedness JDK).

Reply:
You are right that sum of two arbitrary values is not guaranteed to be in the signed positive range. That's why I used @SignednessEither which means that value may be interpreted as @Signed or @Unsigned.

Reply:
The Signedness Checker issued no warning because-
I have seen previously also that the Signedness Checker only runs on methods signatures and not on its body of the annotated JDK. This same type of thing happened  earlier too [here](https://github.com/typetools/checker-framework/pull/2400#discussion_r279038468) where I annotated the body but nothing was happening due to that change.

Reply:
1. Can you explain why the result of every call to `sum` can be interpreted as either `@Signed` or `@Unsigned`?
2. Other type systems issue a warning regarding annotations in method bodies.  Please confirm that fact, and determine why the Signedness Checker is not.

Thanks!

Reply:
According to Signedness Checker definition-

1. If a number is @Signed and within a positive signed range then its annotation is replaced by 
    @SignednessEither else @Signed.
2. If a number is @Unsigned and within a positive signed range then its annotation is replaced by 
    @SignednessEither else @Unsigned.

Therefore,
1. If the sum is of two unsigned numbers, it can be in signed positive range or maybe not, 
    therefore, @SignednessEither, which should control both.
2. If the sum is of two signed numbers, it can be in signed positive range or maybe not, 
    therefore, @SignednessEither, which should control both.

Reply:
I think you may have a mistaken understanding of `@SignednessEither`.

`@SignednessEither` is for expressions for which both signed and unsigned interpretations are definitely legal.  That is not the case for the result of `sum`.  Not every call to `sum` is definitely sensible to interpret as unsigned, and not every call is definitely sensible to interpret as unsigned.